app/src/main/java/com/cornellappdev/transit/ui/viewmodels/HomeViewModel.kt

```diff
@@ -58,10 +70,39 @@ class HomeViewModel @Inject constructor(
     }
 
     /**
-     * Change the query in the search bar
+     * True if the stop can be searched via the [query] string
+     */
+    private fun fulfillsQuery(stop: Stop, query: String): Boolean {
+        return !query.isBlank() && stop.name.lowercase().contains(query.lowercase())
+    }
+
+    /**
+     * Change the query in the search bar and update search results
      */
     fun onQueryChange(query: String) {
         searchQuery.value = query;
+        viewModelScope.launch {
+            stopFlow.combine(searchQuery) { allStops, filter ->
+                when (allStops) {
+                    is ApiResponse.Error -> {
+                        emptyList()
+                    }
+
+                    is ApiResponse.Pending -> {
+                        emptyList()
+                    }
+
+                    is ApiResponse.Success -> {
+                        allStops.data.filter { stop ->
+                            fulfillsQuery(stop, filter)
+                        }
+                    }
+                }
+
+            }.collect { list ->
+                _queryFlow.value = list
+            }
+        }
```

Comment:
Ohh wait ok I see what u did here it's kinda funny cuz like it makes sense intuitively BUT this is not the way to use flow combine.

What you should do instead is:

```kotlin
_queryFlow = stopFlow.combine(searchQuery) { allStops, filter ->
    when (allStops) {
        is ApiResponse.Error -> {
            emptyList()
        }

        is ApiResponse.Pending -> {
            emptyList()
        }

        is ApiResponse.Success -> {
            allStops.data.filter { stop ->
                fulfillsQuery(stop, filter)
            }
        }
    }
}
```

This defines `_queryFlow` AS the combination of the two flows that automatically updates with new emits whenever one of the queries or stops responses changes.

The whole point of `.combine()` is so that we DON'T NEED TO EMIT. The combine will do all that hard work for us.

This might seem unintuitive, because your function here will just be **one line**--all it will do is set the `searchQuery` flow value. Because of the combination, this will trigger a re-flow down `_queryFlow`.

Reply:
^ To reiterate, this does not go in the function. This is how you initially define `_queryFlow`. Your function here should be a 1-line implementation.